examples/class-definition/class_definition.bal

```diff
@@ -1,30 +1,126 @@
+    // type a union with a different type, in order to set an initial value to the field.
+    // Here, the `parent` field does so by defining the field to be of type `Person` or nil.
+    public Person? parent = ();
+
+    // The `init` method should initialize any field that does not have an initalizer
+    // expression. If the `init` method accepts parameters, arguments have to be passed
+    // for parameters when using the `new` expression with the class.
+    function init(string name, int age, boolean employed) {
+        // Fields of the class are accessed using `self`.
+        self.name = name;
+        self.age = age;
+        self.employed = employed;
+    }
+
+    // Methods can access all the fields of the class.
+    function isEmployed() returns boolean => self.employed;
+}
+
+// Define a class named `Person`. It contains `public`, `private`, and module-level visible fields.
+// Initializer expressions are specified for all of the fields and this class does not contain
+// an `init` method.
+class Employee {
     public string name = "";
     public int age = 0;
-    // If you need to have a field with its own type, you need to make that field defaultable
-    // (i.e., able to specify a default value).
-    // Here, the `parent` field is made defaultable by allowing it to have nil as a value.
-    public Person? parent = ();
-    // Private fields are only visible within the object and its functions.
     private string email = "default@abc.com";
-    // Protected fields (i.e., no access modifiers) are visible only within the same module.
-    string address = "No 20, Palm grove";
+    string address = "No 20, Palm Grove";
+
+    // Private methods can only be called from within the class;
+    private function getDomainName() returns string => "abc.com";
+
+    // Methods can update fields.
+    // Module-level visible methods can only be called from within the same module.
+    function setEmail(string username) {
+        self.email = string `${username}@${self.getDomainName()}`;
+    }
+
+    // Public methods can be called from anywhere.
+    public function getEmail() returns string => self.email;
+}
+
+class Teenager {
+    string name;
+    int age;
+
+    // An `init` method may return an error.
+    function init(string name, int age) returns error? {
+        if age > 18 {
+            return error(string `${name} is not a teenager!`);
+        }
+
+        self.name = name;
+        self.age = age;
+    }
 }
 
 public function main() {
-    // There are three ways to initialize a `Person` object.
-    Person p1 = new;
-    io:println(p1.age);
+    // If there is no `init` method or the `init` method does not accept parameters,
+    // a value can be constructed with just the `new` expression.
+    Employee e1 = new;
+    io:println(e1.age);
+
+    // Such a value can also be constructed with the `new` expression with empty parentheses.
+    Employee e2 = new ();
```

Comment:
Don't we usually format this type of new expression as `new()`?

Reply:
No, I think we decided to leave a space for consistency with `new Employee()`.